doc/OnlineDocs/explanation/analysis/alternative_solutions.rst

```diff
@@ -2,233 +2,359 @@
+
+For example, we can generate all optimal solutions as follows:
 
 .. doctest::
    :skipif: not glpk_available
 
-   >>> import pyomo.environ as pyo
-   >>> import pyomo.contrib.alternative_solutions as aos
+   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=100, solver="glpk", abs_opt_gap=0.0)
+   >>> print( [soln.objective().value for soln in solns] )
+   [70.0, 70.0]
 
-   >>> values = [10,9,2,1,1]
-   >>> weights = [10,9,2,1,1]
+Similarly, we can generate the six solutions within 40 of the optimum:
 
-   >>> K = len(values)
-   >>> capacity = 12
+.. doctest::
+   :skipif: not glpk_available
+
+   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=100, solver="glpk", abs_opt_gap=40.0)
+   >>> print( [soln.objective().value for soln in solns] )
+   [70.0, 70.0, 60.0, 60.0, 50.0, 30.0]
 
-   >>> m = pyo.ConcreteModel()
-   >>> m.x = pyo.Var(range(K), within=pyo.Binary)
-   >>> m.o = pyo.Objective(expr=sum(values[i] * m.x[i] for i in range(K)), sense=pyo.maximize)
-   >>> m.c = pyo.Constraint(expr=sum(weights[i] * m.x[i] for i in range(K)) <= capacity)
 
-   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=10, solver="glpk", abs_opt_gap = 0.0)
-   >>> assert(len(solns) == 4)
+Pyomo Solution Pools
+--------------------
 
-In this example, we only get the four ``Solution`` objects that have an
-``objective_value`` of 12.  Note that while we wanted only those four
-solutions with no optimality gap, using a gap of half the smallest value
-(in this case .5) will return the same solutions and avoids any machine
-precision issues.
+The *alternative-solutions library* uses solution pools to filter and store solutions generated by an optimizer.
+The following types of solution pools are currently supported:
+
+* ``keep_all`` : This pool stores all solutions. No solutions are filtered out.
+
+* ``keep_latest`` : This pool stores the latest ``max_pool_size`` solutions that are added to the pool.
+
+  * ``max_pool_size`` (non-negative integer) : The maximum number of solutions that are stored.
+
+* ``keep_latest_unique`` : This pool stores the latest ``max_pool_size`` unique solutions that are added to the pool.
+
+  * ``max_pool_size`` (non-negative integer) : The maximum number of solutions that are stored.
+
+* ``keep_best`` : This pool stores the best solutions added to the pool.
+
+  * ``max_pool_size`` (non-negative integer) : The maximum number of solutions that are stored.
+
+  * ``objective`` (function) : A user-specified function that computes the objective value used for comparisons.
+    
+  * ``abs_tolerance`` (non-negative float) : The absolute tolerance that is used to filter solutions.
+
+  * ``rel_tolernace`` (non-negative float) : The relative tolerance that is used to filter solutions.
+
+  * ``sense_is_min`` (bool) : If True, then the pool will keep solutions with the minimal objective values.
```

Comment:
Just for the sake of API consistency, I would vote for calling this `sense` and having it accept values from the `ObjectiveSense` Enum in pyomo.core.